pp_ctl.c

```diff
@@ -4762,27 +4761,23 @@ PP(pp_entereval)
 
 PP(pp_leaveeval)
 {
+    dSP;
     SV **oldsp;
     U8 gimme;
     PERL_CONTEXT *cx;
     OP *retop;
-    int failed;
+    int failed = 0;
```

Comment:
As C99 permits we can make this a `bool`, we can also move it lower down to just before it's first needed.

Reply:
Good point. Done.

Reply:
That isn't correct, it should not be a bool, it is an int. failed is the 'action' parameter to S_pop_eval_context_maybe_croak().